src/wallet/rpcwallet.cpp

```diff
@@ -93,7 +95,18 @@ void WalletTxToJSON(const CWalletTx& wtx, UniValue& entry)
         entry.push_back(Pair("blockindex", wtx.nIndex));
         entry.push_back(Pair("blocktime", mapBlockIndex[wtx.hashBlock]->GetBlockTime()));
         entry.push_back(Pair("expiryheight", (int64_t)wtx.nExpiryHeight));
+        status = "mined";
     }
+    else
+    {
+        const int height = chainActive.Height();
```

Comment:
Is this locked on `cs_main`?

Reply:
Yes i think so.

`WalletTxToJSON` is used directly in 1 rpc call: 
- `gettransaction`, Locked with `LOCK2(cs_main, pwalletMain->cs_wallet);`

Then it is used as part of `ListTransactions`: This is used in 3 more calls: 
- `listtransactions`, locked with `    LOCK2(cs_main, pwalletMain->cs_wallet);`
- `listsinceblock` - locked with `LOCK2(cs_main, pwalletMain->cs_wallet);`.
- `gettransaction` - locked with `    LOCK2(cs_main, pwalletMain->cs_wallet);`